packages/framework/src/types/infer.types.ts

```diff
@@ -0,0 +1,12 @@
+import { Workflow } from './discover.types';
+
+export type InferWorkflowId<T_Workflow extends Workflow> = T_Workflow extends Workflow<infer Id, never> ? Id : never;
+
+export type InferWorkflowPayload<
```

Comment:
This is a lovely DX improvement, but I feel we need to add the `BulkEventParams` and `typedBulkTrigger` utilities to the framework to maximize its DX benefits.

The instructions we give Novu users feel more convoluted than hardcoding a `workflowId` in a constant and reusing it in their code.

Alternatively, how about adding a triggerBulk function on the workflow object?

Reply:
I would suggest we close this PR for the time being until the isomorphic `@novu/api` package is available and ready for close integration with `@novu/framework`. Then, everything can be bundled together in a much simpler way for user-land consumption.